### PR TITLE
Improvement + Fix: Barn Fishing Timer

### PR DESCRIPTION
**Pull request by ItsEmpa**
## What
Refactors Barn Fishing Timer to use MobDetection, adds a warning for when you reach a certain amount of sea creatures, and makes the time actually based on the time each mob has existed for instead of the time the timer has been active for.

This pr now also makes changes to the barn fishing timer to work with the new fishing cap changes Hypixel implemented.

## Changelog Improvements
+ Improved Barn Fishing Timer. - Empa
    * Modified the warning thresholds for Crimson Isle and Crystal Hollows.
    * The timer now only takes into account the user's own sea creatures.
    * Added a warning when reaching a certain number of sea creatures.
## Changelog Fixes
+ Fixed Squid Pet counting towards the Fishing Timer. - Empa

### CODE REVIEW

**Comment by ItsEmpa** [comment]
the scope of this pr has changed to also implement the sea creature cap changes added by hypixel

File: src/main/java/at/hannibal2/skyhanni/features/fishing/FishingTimer.kt [hunk 1]
```diff
@@ -3,126 +3,220 @@ package at.hannibal2.skyhanni.features.fishing
...
 import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
-import kotlin.time.Duration.Companion.milliseconds
 import kotlin.time.Duration.Companion.seconds
 
 @SkyHanniModule
 object FishingTimer {
 
     private val config get() = SkyHanniMod.feature.fishing.barnTimer
     private val barnLocation = LorenzVec(108, 89, -252)
+    private val mobMap = mutableMapOf<Mob, SimpleTimeMark>()
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/features/fishing/FishingTimer.kt]: what does the value represent? please use a more descriptive name for the map

File: src/main/java/at/hannibal2/skyhanni/features/fishing/FishingTimer.kt [hunk 2]
```diff
@@ -3,126 +3,220 @@ package at.hannibal2.skyhanni.features.fishing
...
+            recentMobs.clear()
+            lastNameFished = null
         }
+        updateInfo()
+    }
 
-        currentCount = newCount
-        if (newCount == 0) {
-            startTime = SimpleTimeMark.farPast()
+    private fun handleBabySlugs() {
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/features/fishing/FishingTimer.kt:140]: why do we need to handle baby slugs differently? can this get moved to mob detection?

File: src/main/java/at/hannibal2/skyhanni/features/fishing/FishingTimer.kt [hunk 3]
```diff
@@ -3,126 +3,220 @@ package at.hannibal2.skyhanni.features.fishing
...
     @SubscribeEvent
     fun onRenderOverlay(event: GuiRenderEvent.GuiOverlayRenderEvent) {
-        if (!LorenzUtils.inSkyBlock) return
-        if (!config.enabled) return
+        if (!isEnabled()) return
         if (!rightLocation) return
         if (currentCount == 0) return
         if (!FishingAPI.isFishing()) return
 
+        val text = createDisplay()
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/features/fishing/FishingTimer.kt]: Don't create the text each frame. rather only 10 times a second and cache the display string.

File: src/main/java/at/hannibal2/skyhanni/features/fishing/FishingTimer.kt [hunk 4]
```diff
@@ -3,126 +3,220 @@ package at.hannibal2.skyhanni.features.fishing
...
+        config.pos.renderString(text, posLabel = "BarnTimer")
+    }
+
+    private fun createDisplay(): String {
         val passedSince = startTime.passedSince()
-        val barnTimerAlertTime = (config.alertTime * 1_000).milliseconds
-        val color = if (passedSince > barnTimerAlertTime) "§c" else "§e"
+        val timeColor = if (passedSince > config.alertTime.seconds) "§c" else "§e"
         val timeFormat = passedSince.format(TimeUnit.MINUTE)
+        val countColor = if (config.fishingCapAlert && currentCount >= currentCap.getValue()) "§c" else "§e"
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/features/fishing/FishingTimer.kt]: getValue() is deprecated

**Review by hannibal002 — CHANGES_REQUESTED** [review]
i dont understand why we process magma slugs differently. also the custom handlinb of fishing mobs should maybe get moved into another api class, so that only the count and add logic remains in this class. (extract complexitiy that might be duplicated in future other features into abstract classes early)

File: src/main/java/at/hannibal2/skyhanni/features/fishing/FishingTimer.kt (hunk 2 above)
Comment (hannibal002) [reply src/main/java/at/hannibal2/skyhanni/features/fishing/FishingTimer.kt:140]: discussed on discord

**Review by hannibal002 — APPROVED** [review]

**Pull request merged**